Seems reasonable - I imagine objects with `_ipython_display_` are not intending it for the terminal.

Test failure:

```
======================================================================
FAIL: Objects with _ipython_display_ defined bypass other formatters
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.0/lib/python3.6/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/ipython/ipython/IPython/core/tests/test_formatters.py", line 419, in test_ipython_display_formatter
    nt.assert_equal(d, {})
AssertionError: {'text/plain': '<IPython.core.tests.test_f[80 chars]d0>'} != {}
```

--- a/IPython/core/tests/test_formatters.py
+++ b/IPython/core/tests/test_formatters.py
@@ -407,6 +407,9 @@ def test_ipython_display_formatter():
         def _ipython_display_(self):
             raise NotImplementedError
     
+    save_enabled = f.ipython_display_formatter.enabled
+    f.ipython_display_formatter.enabled = True
+    
     yes = SelfDisplaying()
     no = NotSelfDisplaying()
     
@@ -419,6 +422,9 @@ def test_ipython_display_formatter():
     nt.assert_equal(d, {})
     nt.assert_equal(md, {})
     nt.assert_equal(catcher, [yes])
+
+    f.ipython_display_formatter.enabled = save_enabled
+
 
 def test_json_as_string_deprecated():
     class JSONString(object):

